Lib/test/test_free_threading/test_type.py

```diff
@@ -95,6 +96,32 @@ def reader_func():
 
         self.run_one(writer_func, reader_func)
 
+    def test___class___modification(self):
```

Comment:
How long does this test usually take? If it's longer than about 10 seconds, we should mark it as requiring the CPU resource (which isn't strictly true, but it's the resource we normally use for "takes a long time").

(You might need to import `support`).

```suggestion
    @support.requires_resource('cpu')
    def test___class___modification(self):
```

This test is probably better off in `test_super` with the other `__class__` tests, and deserves a comment saying that without an audit hook set, it isn't really testing anything, but normal test runs should have a hook set.

Reply:
Makes sense. Thanks for the review! Btw we need to land this together with https://github.com/python/cpython/pull/120165, because this test might actually crash the free-threaded build.

Reply:
It takes 3.3s with a debug build with TSAN on my computer. So that means it's probably plenty fast on normal computers.

Reply:
106ms on debug build on my computer, with some modifications. That's good enough!